operators/pkg/controller/apmserverelasticsearchassociation/apmserverelasticsearchassociation_controller.go

```diff
@@ -164,14 +165,18 @@ func (r *ReconcileApmServerElasticsearchAssociation) Reconcile(request reconcile
 	}
 
 	newStatus, err := r.reconcileInternal(apmServer)
-	// maybe update status
-	origStatus := apmServer.Status.DeepCopy()
-	apmServer.Status.Association = newStatus
-
-	if !reflect.DeepEqual(*origStatus, apmServer.Status) {
+	oldStatus := apmServer.Status.Association
+	if !reflect.DeepEqual(oldStatus, newStatus) {
+		apmServer.Status.Association = newStatus
 		if err := r.Status().Update(&apmServer); err != nil {
 			return defaultRequeue, err
 		}
+		r.recorder.AnnotatedEventf(&apmServer,
+			annotation.ForAssociationStatusChange(oldStatus, newStatus),
```

Comment:
Curious about why we choose to use annotations here?

Reply:
Mainly for machine use. It would be easier for a monitoring system (and tests) to figure out the state transition by reading the annotations rather than parsing the message text.